src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -266,14 +267,12 @@ public void getDocIds(DocIdPusher pusher) throws IOException,
   void addMetadataToRecordBuilder(DocIdPusher.Record.Builder builder,
       ResultSet rs) throws SQLException {
     ResultSetMetaData rsMetaData = rs.getMetaData();
-    int numberOfColumns = rsMetaData.getColumnCount();
-    for (int i = 1; i < (numberOfColumns + 1); i++) {
-      String columnName = rsMetaData.getColumnLabel(i);
-      Object value = rs.getObject(i);
-      String key = metadataColumns.getMetadataName(columnName);
-      if (key != null) {
-        builder.addMetadata(key, "" + value);
-      }
+    MetadataColumns metadataCols = (metadataColumns == null)
+        ? new MetadataColumns(rsMetaData) : metadataColumns;
+    for (String columnName : metadataCols.keySet()) {
+      int index = rs.findColumn(columnName);
```

Comment:
We should add a test with an unmatched column that throws a SQLException here.

Now I'm nervous about throwing here. Instead of silently skipping columns we'll kill the whole lister or retriever call. It's a config error, true. For now, let's just add a TODO to try to verify the columns in the map earlier.